sequencefile/reader.go

```diff
@@ -0,0 +1,177 @@
+package sequencefile
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"io/ioutil"
+)
+
+// A Reader reads key/value pairs from an input stream.
+type Reader struct {
+	Header          Header
+	syncMarkerBytes []byte
+
+	reader io.Reader
+	closed bool
+	err    error
+
+	buf   bytes.Buffer
+	key   []byte
+	value []byte
+}
+
+// New returns a new Reader for sequencefiles, reading input from r. If the
+// io.Reader is positioned at the start of a file, you should immediately call
+// ReadHeader to read through the header.
```

Comment:
I dislike APIs that require me to do things in a certain sequence. Looking at you `http.Request.ParseForm`.